package/base-files/files/lib/upgrade/stage2

```diff
@@ -123,15 +123,25 @@ kill_remaining() { # [ <signal> [ <loop> ] ]
 
 indicate_upgrade
 
+skip_services="dnsmasq log network"
+for service in /etc/init.d/*; do
+	service=${service##*/}
+
+	case "$skip_services" in
```

Comment:
You should add leading and trailing spaces around `" $skip_services "` and `*" $service "*` to avoid overmatching (e.g. `skip_services="foobar example baz"` matching `service=bar`

Reply:
ah ok, I was wondering about that because I noticed no difference either way